### PR TITLE
security: Support proxy with rate limiting and include CI test coverage for nginx rev proxy

### PR DESCRIPTION
**Pull request by webzwo0i**
It's ugly but works

### CODE REVIEW

File: src/node/handler/PadMessageHandler.js [hunk 1]
```diff
@@ -181,10 +181,11 @@ exports.handleMessage = async function(client, message)
   var env = process.env.NODE_ENV || 'development';
 
   if (env === 'production') {
+    const clientIPAddress = (settings.trustProxy && client.handshake.headers['x-forwarded-for']) ? client.handshake.headers['x-forwarded-for'] : client.handshake.address;
```
Comment (rhansen) [inline src/node/handler/PadMessageHandler.js:184]:
```suggestion
    const clientIPAddress = remoteAddress[client.id];
```

**Comment by JohnMcLear** [comment]
Sorry dude I just pushed package-lock, can you please resolve that conflict?

File: src/node/handler/PadMessageHandler.js (hunk 1 above)
Comment (webzwo0i) [reply src/node/handler/PadMessageHandler.js]: Perfect. I saw this, but did not realize it's set in the connection setup phase and is available to later socket.io messages.

**Comment by rhansen** [comment]
I rebased the `fix-rate-limit` branch onto `develop`.

**Pull request merged**